ntoskrnl/mm/section.c

```diff
@@ -1263,14 +1261,35 @@ MmMakeSegmentResident(
             {
                 /* Dirtify it if it's a resident page and we're asked to */
                 if (SetDirty && !IS_SWAP_FROM_SSE(Entry))
-                    MmSetPageEntrySectionSegment(Segment, &CurrentOffset, DIRTY_SSE(Entry));
+                {
+                    NT_VERIFY(NT_SUCCESS(MmSetPageEntrySectionSegment(Segment, &CurrentOffset, DIRTY_SSE(Entry))));
+                }
                 continue;
             }
 
             ToReadPageBits |= 1UL << ((ChunkOffset - RangeStart) >> PAGE_SHIFT);
 
             /* Put a wait entry here */
-            MmSetPageEntrySectionSegment(Segment, &CurrentOffset, MAKE_SWAP_SSE(MM_WAIT_ENTRY));
+            Status = MmSetPageEntrySectionSegment(Segment, &CurrentOffset, MAKE_SWAP_SSE(MM_WAIT_ENTRY));
```

Comment:
Why would this one need a rollback, but not for example, the one just above (line 1265)?

Reply:
The one in line 1265 cannot fail (at least it should never fail), because a previous call to `MmGetPageEntrySectionSegment` returned non-NULL, i.e. the entry already exists. In this path `MmGetPageEntrySectionSegment` returned NULL, so there is no entry yet, and we must allocate one, which can fail, if non-paged pool is low.